File: Benchmarks/enterprise/python/flask/testcode/benchmark_test_18084.py
# SPDX-License-Identifier: Apache-2.0
from markupsafe import Markup
import os


def BenchmarkTest18084():
    env_value = os.environ.get('USER_INPUT', '')
    if env_value:
        data = env_value
    else:
        data = ''
    return Markup('<div>' + str(data) + '</div>')

File: Benchmarks/enterprise/java/spring/testcode/BenchmarkTest18084.java
// SPDX-License-Identifier: Apache-2.0
import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;
import org.springframework.web.bind.annotation.*;

@RestController
public class BenchmarkTest18084 {

    @PostMapping(path="/BenchmarkTest18084", consumes="text/plain")
    public void BenchmarkTest18084(@RequestBody String rawBody, HttpServletRequest request, HttpServletResponse response) throws Exception {
        String rawData = rawBody != null ? rawBody : "";
        String data = "" + rawData;
        if ("admin".equals(data)) {
            response.getWriter().print("{\"role\":\"admin\"}");
            return;
        }
        response.sendError(403, "forbidden");
    }
}
